File: src/main/java/hilfling/backend/hilfing/model/PurchaseOrder.java
package hilfling.backend.hilfing.model;

import lombok.Data;
import org.hibernate.annotations.ColumnDefault;

import javax.persistence.*;
import javax.validation.constraints.Email;
import java.io.Serializable;
import java.util.Date;


@Data
@Entity
@Table(name = "purchase_order")
public class PurchaseOrder implements Serializable, BaseModel {
    public PurchaseOrder() {
    }

    public PurchaseOrder(
            String name,
            @Email String email,
            String address,
            String zipCode,
            String city,
            Boolean sendByPost,
            String comment,
            Date dateCreated,
            Boolean completed
    ) {
        this.name = name;
        this.email = email;
        this.address = address;
        this.zipCode = zipCode;
        this.city = city;
        this.sendByPost = sendByPost;
        this.comment = comment;
        this.dateCreated = dateCreated;
        this.completed = completed;
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name", nullable = false)
    private String name;

    @Email
    @Column(name = "email", nullable = false)
    private String email;

    @Column(name = "address", nullable = false)
    private String address;

    @Column(name = "zip_code", nullable = false)
    private String zipCode;

    @Column(name = "city", nullable = false)
    private String city;

    @Column(name = "send_by_post", nullable = false)
    private Boolean sendByPost;

    @Column(name = "comment", columnDefinition = "TEXT")
    private String comment;

    @Column(name = "date_created", nullable = false)
    private Date dateCreated;

    @Column(name = "completed", nullable = false)
    @ColumnDefault("false")
    private Boolean completed;

}

File: src/main/kotlin/no/fg/hilflingbackend/model/PurchaseOrder.kt
package no.fg.hilflingbackend.model

import org.ktorm.database.Database
import org.ktorm.entity.Entity
import org.ktorm.entity.sequenceOf
import org.ktorm.schema.boolean
import org.ktorm.schema.varchar

interface PurchaseOrder : BaseModel<PurchaseOrder> {
  companion object : Entity.Factory<PurchaseOrder>()

  var name: String
  var email: String
  var address: String
  var zipCode: String
  var city: String
  var sendByPost: Boolean
  var comment: String
  var isCompleted: Boolean
}

object PurchaseOrders : BaseTable<PurchaseOrder>("purchase_order") {
  val name = varchar("name").bindTo { it.name }
  val email = varchar("email").bindTo { it.email }
  val address = varchar("address").bindTo { it.address }
  val zipCode = varchar("zip_code").bindTo { it.zipCode }
  val city = varchar("city").bindTo { it.city }
  val sendByPost = boolean("send_by_post").bindTo { it.sendByPost }
  val comment = varchar("comment").bindTo { it.comment }
  val isCompleted = boolean("is_completed").bindTo { it.isCompleted }
}

val Database.purchase_orders get() = this.sequenceOf(PurchaseOrders)
